### ISSUE TITLE
Mark asm code with no-executable-stack fails to build on M1 mac

### ISSUE THREAD

**Issue by srmainwaring** [ros2/Mimick#27]
[#26](https://github.com/ros2/Mimick/pull/26) is not compatible with M1 macs.

System: macOS Ventura 13.4
Machine: MacBook Pro Apple M1 Max

Repeating details from https://github.com/ros2/Mimick/pull/26#issuecomment-1731130663:

```bash
% clang --version
Apple clang version 14.0.3 (clang-1403.0.22.14.1)
Target: arm64-apple-darwin22.5.0
Thread model: posix
InstalledDir: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin
```

```bash
[  4%] Building ASM object CMakeFiles/mimick.dir/src/asm/trampoline-aarch64.S.o
/Users/rhys/Code/ros2/rolling/ros2-rolling/src/Mimick/src/asm/trampoline-aarch64.S:80:19: error: unexpected token in '.section' directive
.section .note.GNU-stack, "", @progbits
```

**Comment by PhDittmann** [comment]
Adding `#ifdef __ELF__` should fix this, see #28.

**Comment by mjcarroll** [comment]
Closed by #28
